dbms/src/Interpreters/Join.cpp

```diff
@@ -1926,40 +1993,38 @@ void Join::joinBlock(Block & block) const
 
     std::shared_lock lock(rwlock);
 
-    checkTypesOfKeys(block, sample_block_with_keys);
```

Comment:
Why move this check to `HashJoinProbeInputStream`

Reply:
This is because it avoids multiple calls to the `checkTypesOfKeys`.